packages/medusa/src/services/__tests__/csv-parser.js

```diff
@@ -0,0 +1,130 @@
+      products = [
+        {
+          title: "T-shirt",
+          subtitle: "summer tee",
+        },
+        {
+          title: "Sunglasses",
+          subtitle: "Red sunglasses",
+        },
+      ]
+
+    beforeEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it("given a readable stream, can parse the stream content", async () => {
+      const stream = Readable.from(csvContent)
+      const content = await csvParser.parse(stream)
+
+      expect(content).toEqual(products)
+    })
+  })
+
+  describe("buildData", () => {
+    describe("schema validation", () => {
+      const csvParser = new CsvParser(createContainer(), {
+        columns: [
+          {
+            name: "title",
+            validator: {
+              validate: async (value) => {
+                if (/\d/.test(value)) {
+                  throw new Error("title should not contain a number")
+                }
+              },
+            },
+          },
+        ],
+      })
+
+      it("given a line containing a column which is not defined in the schema, then validation should fail", async () => {
+        try {
+          await csvParser.buildData([{ first_name: "lebron " }])
+        } catch (err) {
+          expect(err.message).toEqual(
+            "Unable to to treat column first_name from the csv file. No target column found in the provided schema"
+          )
+        }
+      })
+
+      it("given a line containing a column which does not pass a validation constraint, then validation should fail", async () => {
+        try {
+          await csvParser.buildData([{ title: "contains a number 1" }])
+        } catch (err) {
+          expect(err.message).toEqual("title should not contain a number")
+        }
+      })
+
+      it("given a line which passes all validation constraints, then should returned validated content", async () => {
+        const content = await csvParser.buildData([{ title: "great product" }])
+
+        expect(content).toEqual([
+          {
+            title: "great product",
+          },
+        ])
+      })
+    })
+
+    describe("mapTo", () => {
+      const csvParser = new CsvParser(createContainer(), {
+        columns: [
+          {
+            name: "title",
+            mapTo: "product_title",
+          },
+        ],
+      })
+
+      it("given a mapTo field for a column, when building data including that column, should rename the column name to what mapTo refers to", async () => {
+        const content = await csvParser.buildData([{ title: "a product" }])
+
+        expect(content).toEqual([
+          {
+            product_title: "a product",
+          },
+        ])
+      })
+    })
+
+    describe("transformer", () => {
+      const csvParser = new CsvParser(createContainer(), {
+        columns: [
+          {
+            name: "title",
+          },
+          {
+            name: "price usd",
+            transformer: (value) => Math.round(parseFloat(value) * 100),
```

Comment:
issue:
Here I would suggest to use `Number(value)` instead of `parseFloat(value)`.
I know that based on the name of the method, it seams like the logical one to use, but in reality, here is the result you can expect with each of them:
```ts
parseFloat("3px"); // => 3
parseFloat("9,59 €"); // => 9
// ------------------------------------
Number("3px"); // => NaN
Number("9,59 €"); // => NaN
```
also, the transformer should take care of cleaning the value before, for example to remove any extra space that would lead to wrong transformation. `trim(value)`.